src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/ConfigurationClientExtensions.cs

```diff
@@ -119,7 +119,7 @@ public static async Task<IEnumerable<KeyValueChange>> GetKeyValueChangeCollectio
                 {
                     await foreach(ConfigurationSetting setting in client.GetConfigurationSettingsAsync(selector).ConfigureAwait(false))
                     {
-                        if (!eTagMap.TryGetValue(setting.Key, out string etag) || !etag.Equals(setting.ETag))
+                        if (!eTagMap.TryGetValue(setting.Key, out string etag) || !setting.ETag.Equals(etag))
```

Comment:
>setting.ETag.Equals(etag) [](start = 83, length = 25)

It fixes the issue, but it looks like the issue stems from the fact that we are converting the `ETag` to a string in eTagMap. Should we keep it as an ETag instead and use it as intended for etag comparison?